ld: add page (#7922)
tldr-pages/tldr: pages/common/ld.md

# ld

> > The GNU ELF linker.

Reviewer: I think this is a bit more descriptive.

```suggestion
> Link object files together.
```

> > More information: <https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_mono/ld.html>.

Reviewer: Why exactly did you choose to use 28 year old documentation? I'd prefer using current one:

```suggestion
> More information: <https://sourceware.org/binutils/docs-2.38/ld.html>.
```

> - Link an object file with no dependencies into the executable:

Reviewer 2: ```suggestion
- Link a specific object file with no dependencies into an executable:
```

I think it's better to use `a`/`an` here instead of `the` because we are not talking about specific executable. Everything else LGTM. ;)

> - Link an object file with no dependencies into an executable:

Reviewer 2: ```suggestion
- Link the object file with no dependencies into the executable:
```

Reviewer 3: No, this reads better with `an`, as the executable in question is not a specific executable - this is a generic case of an action.

> `ld {{file.o}} -o {{executable}}`

Author: Thank you, that makes sense. In the last example I have paths to a few common linux libraries, 1) should I avoid those because this is in common/ and if not 2) should I keep the common paths or should I replace them with `{{path/to/...}}`s if that makes sense?

> `ld {{file.o}} -o {{executable}}`

Reviewer 4: If the example is specific for `glibc` then it's OK.

- Link two object files together:

> `ld {{path/to/file.o}} {{path/to/file1.o}} --output {{path/to/executable}}`

Reviewer 4: ```suggestion
`ld {{path/to/file1.o}} {{path/to/file2.o}} --output {{path/to/output_executable}}`
```

> 

Reviewer: Building shared libraries is another common usage, could you insert an example of that here?

Author: on it

> - Dynamically link a 64-bit program to glibc (file paths change depending on system):

Reviewer: ```suggestion
- Dynamically link an x86_64 program to glibc (file paths change depending on the system):
```

> - Dynamically link a 64-bit program to glibc (file paths change depending on system):

Reviewer 2: ```suggestion
- Dynamically link the 64-bit program to glibc (file paths change depending on system):
```

Reviewer 5: Hi, there is a small error here. It is more grammatically correct to use `a` in the context of what is said here, since `64-bit program` is not being defined and is more generic therefore using `a` is more applicable @

Reviewer: I don't know, how you pronounce `64-bit`, but I'd use `a` instead of `an`.

Reviewer 3: This should be left as-is.

Ref https://www.merriam-webster.com/words-at-play/is-it-a-or-an

Reviewer 5: :facepalm: rookie mistake, I don't think I proof-read

Reviewer 5: Actually nevermind, there was a commit to make it x64..... therefore an does work and is correct.

> `ld --output {{executable}} -dynamic-linker /lib/ld-linux-x86-64.so.2 /lib/crt1.o /lib/crti.o -lc {{file.o}} /lib/crtn.o`

Reviewer 4: ```suggestion
`ld --output {{path/to/output_executable}} --dynamic-linker /lib/ld-linux-x86-64.so.2 /lib/crt1.o /lib/crti.o -lc {{path/to/file.o}} /lib/crtn.o`
```
